catroid/src/androidTest/java/org/catrobat/catroid/uiespresso/content/brick/app/VisualPlacementActivityTest.java

```diff
@@ -0,0 +1,118 @@
+ */
+
+package org.catrobat.catroid.uiespresso.content.brick.app;
+
+import android.content.Intent;
+import android.support.test.espresso.Espresso;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.test.utils.TestUtils;
+import org.catrobat.catroid.visualplacement.VisualPlacementActivity;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.matcher.ViewMatchers.withId;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+
+import static org.catrobat.catroid.ui.SpriteActivity.EXTRA_X_TRANSFORM;
+import static org.catrobat.catroid.ui.SpriteActivity.EXTRA_Y_TRANSFORM;
+import static org.catrobat.catroid.uiespresso.content.brick.utils.BrickTestUtils.createProjectAndGetStartScript;
+import static org.catrobat.catroid.visualplacement.VisualPlacementActivity.X_COORDINATE_BUNDLE_ARGUMENT;
+import static org.catrobat.catroid.visualplacement.VisualPlacementActivity.Y_COORDINATE_BUNDLE_ARGUMENT;
+
+@RunWith(AndroidJUnit4.class)
+public class VisualPlacementActivityTest {
+	private static final int XPOS = -200;
+	private static final int YPOS = 500;
+
+	@Rule
+	public final ActivityTestRule<VisualPlacementActivity> baseActivityTestRule =
+			new ActivityTestRule<>(VisualPlacementActivity.class, false, false);
+
+	@Before
+	public void setUp() {
+		createProjectAndGetStartScript(VisualPlacementActivity.class.getSimpleName());
+		Intent i = new Intent();
+		i.putExtra(EXTRA_X_TRANSFORM, XPOS);
+		i.putExtra(EXTRA_Y_TRANSFORM, YPOS);
+		baseActivityTestRule.launchActivity(i);
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		TestUtils.deleteProjects(VisualPlacementActivityTest.class.getSimpleName());
+	}
+
+	@Test
+	public void testResultWhenConfirmClicked() {
+		onView(withId(R.id.confirm)).perform(click());
+
+		Assert.assertTrue(baseActivityTestRule.getActivity().isFinishing());
+		Intent resultIntent = baseActivityTestRule.getActivityResult().getResultData();
+		Assert.assertEquals(XPOS, resultIntent.getExtras().get(X_COORDINATE_BUNDLE_ARGUMENT));
+		Assert.assertEquals(YPOS, resultIntent.getExtras().get(Y_COORDINATE_BUNDLE_ARGUMENT));
+	}
+
+	@Test
+	public void testResultWithClickOnCenter() {
+		onView(withId(R.id.frame_container)).perform(click());
+		onView(withId(R.id.confirm)).perform(click());
+
+		Assert.assertTrue(baseActivityTestRule.getActivity().isFinishing());
+		Intent resultIntent = baseActivityTestRule.getActivityResult().getResultData();
+		Assert.assertEquals(0, resultIntent.getExtras().get(X_COORDINATE_BUNDLE_ARGUMENT));
+		Assert.assertEquals(1, resultIntent.getExtras().get(Y_COORDINATE_BUNDLE_ARGUMENT));
```

Comment:
very nice!!!

if you could just add another test for the spriteActivity that checks that the outgoing intent also contains the right value we would have both sides tested!